File: boss_app/lib/core/api/api_client.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

/// Centralized API client for the Boss App.
/// Manages JWT authentication headers and request scoping.
class ApiClient {
  ApiClient._();
  static final ApiClient instance = ApiClient._();

  final String baseUrl = 'https://mobawi-backend-api.onrender.com';
  final String businessId = 'felixpinski';

  String? _token;

  void setToken(String? token) {
    _token = token;
  }

  String? get token => _token;

  Map<String, String> get headers => {
        'Content-Type': 'application/json',
        'x-business-id': businessId,
        if (_token != null) 'Authorization': 'Bearer $_token',
      };

  Future<http.Response> get(String path) async {
    return await http.get(Uri.parse('$baseUrl$path'), headers: headers);
  }

  Future<http.Response> post(String path, Map<String, dynamic> body) async {
    return await http.post(
      Uri.parse('$baseUrl$path'),
      headers: headers,
      body: jsonEncode(body),
    );
  }

  Future<http.Response> put(String path, Map<String, dynamic> body) async {
    return await http.put(
      Uri.parse('$baseUrl$path'),
      headers: headers,
      body: jsonEncode(body),
    );
  }

  Future<http.Response> delete(String path) async {
    return await http.delete(Uri.parse('$baseUrl$path'), headers: headers);
  }
}

File: mobawi_admin/src/apiClient.js
import axios from 'axios';

const API_URL = import.meta.env.VITE_API_URL || 'https://mobawi-backend-api.onrender.com';

const apiClient = axios.create({
  baseURL: API_URL,
  headers: {
    'Content-Type': 'application/json',
  },
});

apiClient.interceptors.request.use((config) => {
  const token = localStorage.getItem('nexus_token');
  if (token) {
    config.headers.Authorization = `Bearer ${token}`;
  }
  return config;
});

export const login = async (username, password) => {
  const response = await apiClient.post('/api/auth/login', { username, password });
  if (response.data.token) {
    localStorage.setItem('nexus_token', response.data.token);
    localStorage.setItem('nexus_user', JSON.stringify(response.data.user));
  }
  return response.data;
};

export const logout = () => {
  localStorage.removeItem('nexus_token');
  localStorage.removeItem('nexus_user');
  window.location.href = '/login';
};

export const getOverview = async () => {
  const response = await apiClient.get('/api/nexus/overview');
  return response.data;
};

export const getBusinesses = async () => {
  const response = await apiClient.get('/api/admin/businesses');
  return response.data;
};

export const getApplications = async () => {
  const response = await apiClient.get('/api/nexus/applications');
  return response.data;
};

export const suspendBusiness = async (id) => {
  const response = await apiClient.post(`/api/admin/businesses/${id}/suspend`);
  return response.data;
};

export const activateBusiness = async (id) => {
  const response = await apiClient.post(`/api/admin/businesses/${id}/activate`);
  return response.data;
};

export default apiClient;
